internals-js/src/definitions.ts

```diff
@@ -3215,9 +3215,16 @@ function copy(source: Schema, dest: Schema) {
   for (const type of typesToCopy(source, dest)) {
     dest.addType(newNamedType(type.kind, type.name));
   }
+  // Directives can use other directives in their arguments. So, like types, we first shallow copy
+  // directives so future references to any of them can be dereferenced. We'll copy the actual
+  // definition later after all directives are defined.
   for (const directive of directivesToCopy(source, dest)) {
-    copyDirectiveDefinitionInner(directive, dest.addDirectiveDefinition(directive.name));
+    dest.addDirectiveDefinition(directive.name);
   }
+  for (const directive of directivesToCopy(source, dest)) {
```

Comment:
**note** I moved this directly after the calls to `addDirectiveDefinition` and specifically before `copySchemaDefinitionInner`. Putting it after caused a bunch of composition test failures.

Reply:
Yes, it actually make sense. That's because `copySchemaDefinitionInner` needs the definitions to be "complete" (have their argument) to properly handle it's directive applications, and more precisely to recognise/handle the `@link` directive (reminder: `@link` itself can actually be renamed, so detecting it is not based on its name, but rather on checking that the argument matches the spec).

Reply:
That makes sense, I appreciate the clarification